cleanlab/token_classification/summary.py

```diff
@@ -0,0 +1,165 @@
+                % (words[r], freq[r])
+            )
+
+        info = [[word, f] for word, f in zip(words, freq)]
+        info = sorted(info, key=lambda x: x[1], reverse=True)
+        return pd.DataFrame(info, columns=["token", "num_label_issues"])
+
+    if not class_names:
+        print(
+            "Classes will be printed in terms of their integer index since `class_names` was not provided. "
+        )
+        print("Specify this argument to see the string names of each class. \n")
+
+    n = pred_probs[0].shape[1]
+    for issue in issues:
+        i, j = issue
+        word = given_words[i][j]
+        label = labels[i][j]
+        pred = pred_probs[i][j].argmax()
+        if word not in count:
+            count[word] = np.zeros([n, n], dtype=int)
+        if (label, pred) not in exclude:
+            count[word][label][pred] += 1
+    words = [word for word in count.keys()]
+    freq = [np.sum(count[word]) for word in words]
+    rank = np.argsort(freq)[::-1][:top]
+
+    for r in rank:
+        matrix = count[words[r]]
+        most_frequent = np.argsort(count[words[r]].flatten())[::-1]
+        print(
+            "Token '%s' is potentially mislabeled %d times throughout the dataset"
+            % (words[r], freq[r])
+        )
+        if verbose:
+            print(
+                "---------------------------------------------------------------------------------------"
+            )
+            for f in most_frequent:
+                i, j = f // n, f % n
+                if matrix[i][j] == 0:
+                    break
+                if class_names:
+                    print(
+                        "labeled as class `%s` but predicted to actually be class `%s` %d times"
+                        % (class_names[i], class_names[j], matrix[i][j])
+                    )
+                else:
+                    print(
+                        "labeled as class %d but predicted to actually be class %d %d times"
+                        % (i, j, matrix[i][j])
+                    )
+        print()
+    info = []
+    for word in words:
+        for i in range(n):
+            for j in range(n):
+                num = count[word][i][j]
+                if num > 0:
+                    if not class_names:
+                        info.append([word, i, j, num])
+                    else:
+                        info.append([word, class_names[i], class_names[j], num])
+    info = sorted(info, key=lambda x: x[3], reverse=True)
+    return pd.DataFrame(
+        info, columns=["token", "given_label", "predicted_label", "num_label_issues"]
+    )
+
+
+def filter_by_token(token, issues, given_words):
+    returned_issues = []
+    for issue in issues:
+        i, j = issue
+        if token.lower() == given_words[i][j].lower():
+            returned_issues.append(issue)
+    return returned_issues
+
+
+def issues_from_scores(sentence_scores, token_scores, threshold=0.2):
```

Comment:
I think this function belongs in either `rank` or `filter`. Leave which one is better up to you :)

Reply:
How did you decide on the threshold being 0.2?

Reply:
it's really arbitrary...changed to 0.1 for now

Reply:
nice I like the rounder numbers better 😛